### ISSUE TITLE
Remove camlp4

### ISSUE THREAD

**Issue by samoht** [mirage/ocaml-github#104]
Currently, `opam pin add github --dev` fails if you have `js_of_ocaml` installed:

```
#=== ERROR while installing github.dev ========================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml setup.ml -build
# path         /Users/thomas/.opam/system/build/github.dev
# compiler     system (4.02.3)
# exit-code    1
# env-file     /Users/thomas/.opam/system/build/github.dev/github-16474-6cc03f.env
# stdout-file  /Users/thomas/.opam/system/build/github.dev/github-16474-6cc03f.out
# stderr-file  /Users/thomas/.opam/system/build/github.dev/github-16474-6cc03f.err
### stdout ###
# [...]
# /Users/thomas/.opam/system/bin/ocamlfind ocamlc -c -g -annot -bin-annot -principal -strict-sequence -I lib -I unix -I js -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -package lwt -package lwt.unix -package stringext -package uri -package yojson -w @5@8@10@11@12@14@23@24@26@29 -I lib_test -I js -I lib -I unix -o lib_test/parse_events.cmo lib_test/parse_events.ml
# /Users/thomas/.opam/system/bin/ocamlfind ocamlopt -c -g -annot -bin-annot -principal -strict-sequence -I lib -I unix -I js -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -package lwt -package lwt.unix -package stringext -package uri -package yojson -w @5@8@10@11@12@14@23@24@26@29 -I lib_test -I js -I lib -I unix -o lib_test/parse_events.cmx lib_test/parse_events.ml
# /Users/thomas/.opam/system/bin/ocamlfind ocamlopt -g -linkpkg -package atdgen -package cmdliner -package cohttp -package cohttp.lwt -package cohttp.lwt-core -package lwt -package uri -package yojson lib/github_core.cmxa lib/github_t.cmx lib/github_j.cmx lib_test/parse_events.cmx -o lib_test/parse_events.native
# /Users/thomas/.opam/system/bin/ocamlfind ocamldep -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -package lwt -package lwt.unix -package stringext -package uri -package yojson -modules lib_test/test_javascript.ml > lib_test/test_javascript.ml.depends
# + /Users/thomas/.opam/system/bin/ocamlfind ocamldep -package atdgen -package bytes -package cmdliner -package cohttp -package cohttp.js -package cohttp.lwt -package cohttp.lwt-core -package js_of_ocaml -package lwt -package lwt.unix -package stringext -package uri -package yojson -modules lib_test/test_javascript.ml > lib_test/test_javascript.ml.depends
# File "lib_test/test_javascript.ml", line 36, characters 10-12:
# Error: Syntax error: ')' expected
# File "lib_test/test_javascript.ml", line 34, characters 6-7:
# Error: This '(' might be unmatched
# Command exited with code 2.
```

Would be great to fix that by either adding an explicit dep to camlp4 in the `opam` file or to remove `camlp4` dependency (preferred fix)

**Comment by rgrinberg** [comment]
Whoops this is my fault. I will fix this soon if nobody does it.

But I'm curious as to why the tests are even being compiled here.

**Comment by samoht** [comment]
That's a good question. Would be great to fix that as well while you are at it :-)

**Comment by dsheets** [comment]
This was fixed in #106.
